Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/modules/diagnosiskeyprovider/DefaultDiagnosisKeyProvider.kt

```diff
@@ -22,95 +19,37 @@ class DefaultDiagnosisKeyProvider @Inject constructor(
-        configuration: ExposureConfiguration?,
-        token: String
-    ): Boolean {
-        return try {
-            if (keyFiles.isEmpty()) {
-                Timber.d("No key files submitted, returning early.")
-                return true
-            }
-
-            val usedConfiguration = if (configuration == null) {
-                Timber.w("Passed configuration was NULL, creating fallback.")
-                ExposureConfiguration.ExposureConfigurationBuilder().build()
-            } else {
-                configuration
-            }
-
-            if (googleAPIVersion.isAtLeast(GoogleAPIVersion.V16)) {
-                provideKeys(keyFiles, usedConfiguration, token)
-            } else {
-                provideKeysLegacy(keyFiles, usedConfiguration, token)
-            }
-        } catch (e: Exception) {
-            Timber.e(
-                e, "Error during provideDiagnosisKeys(keyFiles=%s, configuration=%s, token=%s)",
-                keyFiles, configuration, token
-            )
-            throw e
+    override suspend fun provideDiagnosisKeys(keyFiles: Collection<File>): Boolean {
+        if (keyFiles.isEmpty()) {
+            Timber.d("No key files submitted, returning early.")
+            return true
         }
-    }
 
-    private suspend fun provideKeys(
-        files: Collection<File>,
-        configuration: ExposureConfiguration,
-        token: String
-    ): Boolean {
-        Timber.d("Using non-legacy key provision.")
-
-        if (!submissionQuota.consumeQuota(1)) {
-            Timber.w("Not enough quota available.")
-            // TODO Currently only logging, we'll be more strict in a future release
-            // return false
+        if (!googleAPIVersion.isAtLeast(GoogleAPIVersion.V15)) {
+            // Actually this shouldn't happen
+            Timber.d("No key files submitted because client uses an old unsupported version")
+            return false
         }
 
-        performSubmission(files, configuration, token)
-        return true
-    }
-
-    /**
-     * We use Batch Size 1 and thus submit multiple times to the API.
-     * This means that instead of directly submitting all files at once, we have to split up
-     * our file list as this equals a different batch for Google every time.
-     */
-    private suspend fun provideKeysLegacy(
-        keyFiles: Collection<File>,
-        configuration: ExposureConfiguration,
-        token: String
-    ): Boolean {
-        Timber.d("Using LEGACY key provision.")
-
-        if (!submissionQuota.consumeQuota(keyFiles.size)) {
-            Timber.w("Not enough quota available.")
-            // TODO What about proceeding with partial submission?
-            // TODO Currently only logging, we'll be more strict in a future release
-            // return false
+        if (!submissionQuota.consumeQuota(1)) {
+            Timber.w("No key files submitted because not enough quota available.")
+            return false
```

Comment:
Previously this was not checked because we didn't want to hide quota issues due to wrong frequency/interval checks on a higher level. If this is armed we won't know that something is failing because it is hitting the quota limit for unexpected reasons. :thinking: Was this discussed with someone?

Reply:
see https://github.com/corona-warn-app/cwa-app-android/pull/1643#discussion_r526281904